Applications/WebPortal/WebApp/library/source/components/country_selection_box.tsx

```diff
@@ -17,26 +17,26 @@ interface Properties {
   /** The event handler called when the selection changes. */
   onChange?: (currency: Nexus.CountryCode) => void;
 
+  /** Determines the size of the element. */
   displaySize: DisplaySize;
 }
 
-interface State {
-  options: any[];
-}
-
 /** Displays a country seleciton box. */
-export class CountrySelectionBox extends React.Component<Properties, State> {
+export class CountrySelectionBox extends React.Component<Properties, {}> {
   public static readonly defaultProps = {
     readonly: false,
     onChange: () => {}
   }
 
   constructor(props: Properties) {
     super(props);
-    this.state = {
-      options: []
-    };
-    this.generateList();
+    this.onChange = this.onChange.bind(this);
+    for(const country of this.props.countryDatabase) {
```

Comment:
In a situation like this you'll need to implement ```willReceiveProps``` or whatever method React uses to indicate that props are being updated.

Reply:
Maybe I should just move this into the render method then?